File: abcs/322/a.php
<?php

[$n,$m] = array_map('intval',explode(" ", trim(fgets(STDIN))));
$s = trim(fgets(STDIN));
$t = trim(fgets(STDIN));

// var_dump($lettersBeginsT);
// var_dump($lettersAfterT);
$lettersBeginsT = substr($t,0,$n);
$lettersAfterT = substr($t,strlen($t)-$n);
$isPre = $s === $lettersBeginsT;
$isSuf = $s === $lettersAfterT;

if ($isPre && $isSuf) {
    echo 0 . PHP_EOL;
} elseif ($isPre && !$isSuf) {
    echo 1 . PHP_EOL;

} elseif ($isSuf && !$isPre) {
    echo 2 . PHP_EOL;

} elseif (!$isPre && !$isSuf) {
    echo 3 . PHP_EOL;

}

File: abcs/322/a.cpp
#include "bits/stdc++.h"
#define rep(i,n) for (int i = 0; i < (int)(n); i++)
using namespace std;

int main()
{
  int n,m;
  string s,t;
  cin>>n>>m>>s>>t;
  string begin = t.substr(0,n);
  string after = t.substr(t.length()-n);
  bool isPre = s == begin;
  bool isSuf = s == after;
      if (isPre && isSuf) {
        cout<<0<<endl;
    } else if (isPre && !isSuf) {
        cout<<1<<endl;;

    } else if (isSuf && !isPre) {
        cout<<(2)<<endl;

    } else if (!isPre && !isSuf) {
        cout<<(3)<<endl;

    }

  return 0;
}
